### PR TITLE
Remove last ksonnet reference and direct dependency 

### LINKED ISSUE #591: Switch to jsonnet-libs/k8s-alpha instead of ksonnet
**What is missing?**

Ksonnet was deprecated over a year ago. It seems one of kube-prometheus' maintainers kept a fork of it updated for newer k8s api versions here: https://github.com/kube-jsonnet/k

However this has been replaced with (hopefully) a final resting place, here: https://github.com/jsonnet-libs/k8s-alpha

**Why do we need it?**

The old ksonnet (which is still used in docs) does not use modern API versions, and this is really starting to show on k8s cluster's v1.16+ where, for example, extensions/v1beta1 is no longer recognized (I ran into this when trying to add ingress support).

**Environment**

* kube-prometheus version: 2c1fc1cc11547ca06a143fce6e430e4f7e0be294

**Anything else we need to know?**:
I've migrated the example jsonnet file to the latest version of k8s-alpha jsonnet lib. I've copied this below. 90% of it is identical, main changes are in the imports, and the location of the various mixins has changed.

```
local k = import "github.com/jsonnet-libs/k8s-alpha/1.18/main.libsonnet";
local pvc = k.core.v1.persistentVolumeClaim;
local networking = k.networking.v1beta1;
local ingress = networking.ingress;
local ingressTLS = networking.ingressTLS;
local ingressRule = networking.ingressRule;
local httpIngressPath = networking.httpIngressPath;

local kp = (import 'kube-prometheus/kube-prometheus.libsonnet') +
           (import 'kube-prometheus/kube-prometheus-eks.libsonnet') + {
  _config+:: {
    namespace: 'monitoring',
  },
  prometheusRules+:: {
    groups+: [
      {
        name: 'example-group',
        rules: [
          {
            record: 'aws_eks_available_ip',
            expr: 'sum by(instance) (awscni_total_ip_addresses) - sum by(instance) (awscni_assigned_ip_addresses) < 10',
          },
        ],
      },
    ],
  },
  prometheus+:: {
    prometheus+: {
      spec+: {
        retention: '30d',
        storage: {
          volumeClaimTemplate:
            pvc.new('') +
            pvc.mixin.spec.withAccessModes('ReadWriteOnce') +
            pvc.mixin.spec.resources.withRequests({ storage: '100Gi' }) +
            pvc.mixin.spec.withStorageClassName('gp2'),
        },
      },
    },
  },
  ingress+:: {
    'prometheus':
      ingress.new('prometheus') +
      ingress.mixin.metadata.withNamespace($.prometheus.prometheus.metadata.namespace) +
      ingress.mixin.metadata.withAnnotations({
        'nginx.ingress.kubernetes.io/force-ssl-redirect': 'true',
        'kubernetes.io/ingress.class': 'nginx-internal',
      }) +
      ingress.mixin.spec.withRules(
        ingressRule.withHost('prometheus.aws-uswest2-hldltdev.inblock.com') +
        ingressRule.mixin.http.withPaths(
          httpIngressPath.mixin.backend.withServiceName($.prometheus.service.metadata.name) +
          httpIngressPath.mixin.backend.withServicePort('web')
        ),
      ) +
      ingress.mixin.spec.withTls(
        ingressTLS.withHosts('prometheus.aws-uswest2-hldltdev.inblock.com') +
        ingressTLS.withSecretName('tls-prometheus')
      ),
    'alertmanager':
      ingress.new('alertmanager') +
      ingress.mixin.metadata.withNamespace($.prometheus.prometheus.metadata.namespace) +
      ingress.mixin.metadata.withAnnotations({
        'nginx.ingress.kubernetes.io/force-ssl-redirect': 'true',
        'kubernetes.io/ingress.class': 'nginx-internal',
      }) +
      ingress.mixin.spec.withRules(
        ingressRule.withHost('alertmanager.aws-uswest2-hldltdev.inblock.com') +
        ingressRule.mixin.http.withPaths(
          httpIngressPath.mixin.backend.withServiceName($.alertmanager.service.metadata.name) +
          httpIngressPath.mixin.backend.withServicePort('web')
        ),
      ) +
      ingress.mixin.spec.withTls(
        ingressTLS.withHosts('alertmanager.aws-uswest2-hldltdev.inblock.com') +
        ingressTLS.withSecretName('tls-alertmanager')
      ),
    'grafana':
      ingress.new('grafana') +
      ingress.mixin.metadata.withNamespace($.prometheus.prometheus.metadata.namespace) +
      ingress.mixin.metadata.withAnnotations({
        'nginx.ingress.kubernetes.io/force-ssl-redirect': 'true',
        'kubernetes.io/ingress.class': 'nginx-internal',
      }) +
      ingress.mixin.spec.withRules(
        ingressRule.withHost('grafana.aws-uswest2-hldltdev.inblock.com') +
        ingressRule.mixin.http.withPaths(
          httpIngressPath.mixin.backend.withServiceName($.grafana.service.metadata.name) +
          httpIngressPath.mixin.backend.withServicePort('web')
        ),
      ) +
      ingress.mixin.spec.withTls(
        ingressTLS.withHosts('grafana.aws-uswest2-hldltdev.inblock.com') +
        ingressTLS.withSecretName('tls-grafana')
      ),
  },
};

{ ['setup/0namespace-' + name]: kp.kubePrometheus[name] for name in std.objectFields(kp.kubePrometheus) } +
{
  ['setup/prometheus-operator-' + name]: kp.prometheusOperator[name]
  for name in std.filter((function(name) name != 'serviceMonitor'), std.objectFields(kp.prometheusOperator))
} +
// serviceMonitor is separated so that it can be created after the CRDs are ready
{ 'prometheus-operator-serviceMonitor': kp.prometheusOperator.serviceMonitor } +
{ ['node-exporter-' + name]: kp.nodeExporter[name] for name in std.objectFields(kp.nodeExporter) } +
{ ['kube-state-metrics-' + name]: kp.kubeStateMetrics[name] for name in std.objectFields(kp.kubeStateMetrics) } +
{ ['alertmanager-' + name]: kp.alertmanager[name] for name in std.objectFields(kp.alertmanager) } +
{ ['prometheus-' + name]: kp.prometheus[name] for name in std.objectFields(kp.prometheus) } +
{ ['prometheus-adapter-' + name]: kp.prometheusAdapter[name] for name in std.objectFields(kp.prometheusAdapter) } +
{ ['grafana-' + name]: kp.grafana[name] for name in std.objectFields(kp.grafana) } +
{ [name + '-ingress']: kp.ingress[name] for name in std.objectFields(kp.ingress) }
```

### PR DESCRIPTION
**Pull request by kakkoyun**
* Format remaining jsonnet files.
* Remove the last ksonnet reference.
* Remove ksonnet from direct dependencies.

Fixes #591

### CODE REVIEW

File: jsonnet/kube-prometheus/kube-prometheus-kubeadm.libsonnet [hunk 1]
```diff
@@ -19,14 +19,14 @@ local service(name, namespace, labels, selector, ports) = {
       'kube-controller-manager-prometheus-discovery',
       'kube-system',
       { 'k8s-app': 'kube-controller-manager' },
-      { 'component': 'kube-controller-manager' },
+      { component: 'kube-controller-manager' },
```
Comment (paulfantom) [inline jsonnet/kube-prometheus/kube-prometheus-kubeadm.libsonnet:22]: nit: keys should be either in quotes or without them, mixing both looks a bit off.

File: jsonnet/kube-prometheus/kube-prometheus-kubeadm.libsonnet [hunk 2]
```diff
@@ -19,14 +19,14 @@ local service(name, namespace, labels, selector, ports) = {
...
       [{ name: 'https-metrics', port: 10257, targetPort: 10257 }]
     ),
     kubeSchedulerPrometheusDiscoveryService: service(
       'kube-scheduler-prometheus-discovery',
       'kube-system',
       { 'k8s-app': 'kube-scheduler' },
-      { 'component': 'kube-scheduler' },
+      { component: 'kube-scheduler' },
```
Comment (paulfantom) [inline jsonnet/kube-prometheus/kube-prometheus-kubeadm.libsonnet:29]: same as above

File: jsonnet/kube-prometheus/kube-prometheus-kubeadm.libsonnet (hunk 1 above)
Comment (paulfantom) [reply jsonnet/kube-prometheus/kube-prometheus-kubeadm.libsonnet:22]: This can be fixed later though.

**Review by paulfantom — APPROVED** [review]
LGTM :shipit:

File: jsonnet/kube-prometheus/kube-prometheus-kubeadm.libsonnet (hunk 1 above)
Comment (kakkoyun) [reply jsonnet/kube-prometheus/kube-prometheus-kubeadm.libsonnet:22]: This is autoformatted by `jsonnetfmt`.

File: jsonnet/kube-prometheus/kube-prometheus-kubeadm.libsonnet (hunk 2 above)
Comment (kakkoyun) [reply jsonnet/kube-prometheus/kube-prometheus-kubeadm.libsonnet:29]: This is autoformatted by `jsonnetfmt`.

**Pull request merged**